### PR TITLE
A few more sanity checks for detecting unzip output that's not a file entry

### PR DESCRIPTION
**Pull request by abadger**
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME

unarchive.py
##### ANSIBLE VERSION

```
devel and 2.1
```
##### SUMMARY

I had an old checkout that didn't have the fix for spaces in filenames so I added some checks.  Rebased and found that most of the checks were already there.  A few weren't however, so I thought I'd PR those as an additional set of checks of whether the line we have is a file entry or not.

Also note that there's a rounding error somewhere in the mtime
comparison code.

### CODE REVIEW

File: files/unarchive.py [hunk 1]
```diff
@@ -314,13 +314,19 @@ def is_unarchived(self):
...
             pcs = line.split(None, 7)
+            if len(pcs) != 8:
+                # Too few fields... probably a piece of the header or footer
+                continue
 
             # Check first and seventh field in order to skip header/footer
             if len(pcs[0]) != 7 and len(pcs[0]) != 10: continue
             if len(pcs[6]) != 15: continue
 
+            if pcs[0][0] not in 'dl-?' or not frozenset(pcs[1:]).issubset('rwxst-'):
```
Comment (dagwieers) [inline files/unarchive.py]: Should this not be `frozenset(pcs[0][1:])` ?

Comment (abadger) [inline files/unarchive.py]: Yep. /me changes

Comment (abadger) [inline files/unarchive.py]: Fixed.

**Comment by gregdek** [comment]
Thanks @abadger. To the current maintainers, @dagwieers, @pileofrogs please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**